tests/imjournal-statefile.sh

```diff
@@ -25,11 +25,22 @@ if [ $? -ne 0 ]; then
         echo "SKIP: failed to put test into journal."
         error_exit 77
 fi
-journalctl -an 200 | fgrep -qF "$TESTMSG"
+
+# sleep 1 to get this test to reliably detect the message
+sleep 1
+
+journalctl -an 200 > /dev/null 2>&1
 if [ $? -ne 0 ]; then
         echo "SKIP: cannot read journal."
         error_exit 77
 fi
+
+journalctl -an 200 | grep -qF "$TESTMSG"
+if [ $? -ne 0 ]; then
+        echo "SKIP: cannot find '$TESTMSG' in journal."
```

Comment:
Here I tried to split a failure to run journalctl from a failure to find the message.

I considered failing the test something other than 77, but I don't know all the details about why that might be a bad idea. :)

Reply:
That's a guard if things go wrong for non-rsyslog related tests. The idea is if we cannot find the message ourself, rsyslog can for sure also not find it. So not checking would create a false positive (actually, it did - that's why the guard is now in place).

Reply:
Hi @rgerhards - thank you. The `sleep 1` line was important here.  Additionally, I should note that I split the `journalctl` guard you mention into two pieces. Assuming we can run `journalctl` at all, the second test closes a risk of a false negative. If we can get the sleep interval correct (and maybe on some platforms it is 5.001 minutes) then the failure of `journalctl` to produce the message to the test is a failure to examine.

Reply:
practice was different: it was always either

1. bug in journal
2. journal was rotated away

I don't like to go back to the myriad of false positives we had with the journal tests around a year or so ago. If journalctl cannot find the message, let's assume it was rotated away.

Reply:
Hello @rgerhards - this code will skip in either case. The difference is that the log will indicate if case 1 or case 2 was more likely to have happened. Are you asking that we not do that differentiation?

Reply:
@jmarckel  got you - overlooked it's just two skip cases